security/application.md

```diff
@@ -82,25 +106,62 @@ their email address is not in a valid format, is for presentation. These
 checks, and more, must be duplicated on the backend. Any attacker can use curl
 to bypass your client-side validations.
 
+### Cookies
+
+Cookies are user-controlled input and, therefore, should be treated with
+suspicion. If possible, don't rely on a cookie.
+
+Cookies can be copied between browsers. Just because a request sends a cookie
+does not mean that the cookie was sent by the user's original browser. It might
+come from curl.
+
+One way to retain control over the cookie data is to sign it using a secret key
+only known by the server. Rails does this for you.
+
+## Logging
+
+Logging is a compromise between having enough data to be able to debug
+a problem and having too much personally-identifying information about a user.
+
+Make sure not to log passwords, credit card numbers, or any other information
+that we do not strictly need. Err on the side of not logging any strings, if
+possible.
+
+In Rails, use the `filter_paramters` configuration setting to remove known
+attributes from the logs.
+
+In addition, if you are logging to a service over a network connection, make
+sure the connection itself is secured using TLS.
+
+## Personally-Identifying Information (PII)
+
+As much as you can, do not touch any information you don't need. Some tricks
+for this:
+
+- Send any credit card data directly to the payment processor from the client.
+  They'll give back a token, which we can store safely.
+- You probably don't need the user's sex, gender, date of birth, middle name,
```

Comment:
Aren't sex and gender synonyms? Maybe one can be removed...

Reply:
Dictionary.com has gender as pertaining to "cultural roles" and sex pertaining to "reproductive functions". I think it's nice to keep both here as a reminder that we probably don't need to know either, for most products.